src/wallet/rpcwallet.cpp

```diff
@@ -3482,7 +3482,8 @@ UniValue z_getoperationstatus_IMPL(const UniValue& params, bool fRemoveFinishedO
 // If input notes are small, we might actually require more than one joinsplit per zaddr output.
 // For now though, we assume we use one joinsplit per zaddr output (and the second output note is change).
 // We reduce the result by 1 to ensure there is room for non-joinsplit CTransaction data.
-#define Z_SENDMANY_MAX_ZADDR_OUTPUTS    ((MAX_TX_SIZE / GetSerializeSize(JSDescription(), SER_NETWORK, PROTOCOL_VERSION)) - 1)
+#define Z_SENDMANY_MAX_ZADDR_OUTPUTS_BEFORE_SAPLING    ((MAX_TX_SIZE_BEFORE_SAPLING / GetSerializeSize(JSDescription(), SER_NETWORK, PROTOCOL_VERSION)) - 1)
+#define Z_SENDMANY_MAX_ZADDR_OUTPUTS    ((MAX_TX_SIZE_AFTER_SAPLING / GetSerializeSize(JSDescription(), SER_NETWORK, PROTOCOL_VERSION)) - 1)
```

Comment:
Should this also have a `AFTER_SAPLING` for clarity, consistency, and to make sure we're changing the code correctly in all places?

Reply:
Looked at the outdated comments. We do refer to this constant in an error message just below where it is defined. This will be incorrect until the upgrade, but perhaps this is not a big concern. Also, it is mentioned in the comment above that this may not be completely accurate in the case where there are multiple small notes.

On a side note, should Z_SENDMANY_MAX_ZADDR_OUTPUTS depend on the size of SpendDescritption and/or OutputDescription rather than JSDescription?

Reply:
Hmm, good point about SpendDescription and OutputDescription; they will need to be included in the sum (not replacing JSDescription), since we want z-addr outputs to be either. I'll go make adjustments (though most of them won't be relevant until we do #3215).